src/firebolt/common/base_cursor.py

```diff
@@ -273,25 +304,38 @@ def _reset(self) -> None:
         self._next_set_idx = 0
         self._query_id = ""
 
-    def _parse_response_headers(self, headers: Headers) -> None:
-        """Parse response and update relevant cursor fields."""
-        update_parameters = headers.get("Firebolt-Update-Parameters")
-        # parse update parameters dict and set keys as attributes
-        if update_parameters:
-            # parse key1=value1,key2=value2 comma separated string into dict
-            param_dict = dict(item.split("=") for item in update_parameters.split(","))
-            # strip whitespace from keys and values
-            param_dict = {
-                key.strip(): value.strip() for key, value in param_dict.items()
-            }
-            for key, value in param_dict.items():
-                if key in SERVER_SIDE_PARAMETERS:
-                    self.parameters[key] = value
-                else:
-                    logger.debug(
-                        f"Unknown parameter {key} returned by the server. "
-                        "It will be ignored."
-                    )
+    def _update_set_parameters(self, parameters: Dict[str, Any]) -> None:
+        # Split parameters into immutable and user parameters
+        immutable_parameters = {
+            key: value
+            for key, value in parameters.items()
+            if key in IMMUATABLE_PARAMETER_LIST
+        }
+        user_parameters = {
+            key: value
+            for key, value in parameters.items()
+            if key not in IMMUATABLE_PARAMETER_LIST
+        }
+
+        self._update_server_parameters(immutable_parameters)
+
+        for key, value in user_parameters.items():
```

Comment:
nit: just do `self._set_parameters.update(user_parameters)`?

Reply:
You're right, forgot Python has this syntax.